### ISSUE TITLE
Can't compile the requests library with Nuitka

### ISSUE THREAD

**Issue by enescantas** [Nuitka/Nuitka#3098]
Before submitting an Issue, please review the
[Issue Guidelines](https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md#submitting-an-issue).

- Please check whether the bug was already reported or fixed.

- Please check out if the develop version of Nuitka works better for you.

  Download source, packages [from here](https://nuitka.net/doc/download.html) where you will also
  find instructions how to do it via pip or git.

If you want to post a problem/bug, to help us understand and resolve your issue please check that
you have provided at least the information below, and discard up to here:

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

2.4.8
Commercial: None
Python: 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:23:07) [MSC v.1927 32 bit (Intel)]
Flavor: CPython Official
Executable: C:\Users\Administrator\Desktop\proje\venv\Scripts\python.exe
OS: Windows
Arch: x86
WindowsRelease: 10
Version C compiler: ~\AppData\Local\Nuitka\Nuitka\Cache\DOWNLO~1\gcc\x86\13.2.0-16.0.6-11.0.1-msvcrt-r1\mingw32\bin\gcc.exe (gcc 13.2.0).

  If you are not providing the full output. The issue cannot be solved and will be considered
  `invalid`. The command outputs more than you think, and we are adding more all the time. We do not
  want to left guessing or trying things out to reproduce your issue.

- How did you install Nuitka and Python

python -m venv venv
.\venv\Scripts\activate
python -m pip install --upgrade pip
python -m pip install -U nuitka

  Did you use pip, anaconda, deb, rpm, from source, git clone, then install into a virtualenv or
  not, this is very important usually and one thing, the above command does not tell us (yet).

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip list -v

Package            Version  Location                                                    Installer
------------------ -------- ----------------------------------------------------------- ---------
certifi            2024.7.4 c:\users\administrator\desktop\proje\venv\lib\site-packages pip
charset-normalizer 3.3.2    c:\users\administrator\desktop\proje\venv\lib\site-packages pip
idna               3.8      c:\users\administrator\desktop\proje\venv\lib\site-packages pip
Nuitka             2.4.8    c:\users\administrator\desktop\proje\venv\lib\site-packages pip
ordered-set        4.1.0    c:\users\administrator\desktop\proje\venv\lib\site-packages pip
pip                24.2     c:\users\administrator\desktop\proje\venv\lib\site-packages pip
psutil             6.0.0    c:\users\administrator\desktop\proje\venv\lib\site-packages pip
PyQt5              5.15.11  c:\users\administrator\desktop\proje\venv\lib\site-packages pip
PyQt5-Qt5          5.15.2   c:\users\administrator\desktop\proje\venv\lib\site-packages pip
PyQt5_sip          12.15.0  c:\users\administrator\desktop\proje\venv\lib\site-packages pip
PyQtWebEngine      5.15.7   c:\users\administrator\desktop\proje\venv\lib\site-packages pip
PyQtWebEngine-Qt5  5.15.2   c:\users\administrator\desktop\proje\venv\lib\site-packages pip
requests           2.32.3   c:\users\administrator\desktop\proje\venv\lib\site-packages pip
setuptools         49.2.1   c:\users\administrator\desktop\proje\venv\lib\site-packages pip
urllib3            2.2.2    c:\users\administrator\desktop\proje\venv\lib\site-packages pip
zstandard          0.23.0   c:\users\administrator\desktop\proje\venv\lib\site-packages pip

- Many times when you get an error from Nuitka, your setup may be special

  Then even a `print("hello world")` program will not work, please try that and report that error
  instead. We do not need a report that says "Tensorflow does not work" when nothing actually works
  for you. You ought to also read the User Manual and check that your setup is actually supported
  already.

- Also supply a Short, Self Contained, Correct, Example

import requests

response = requests.get('https://www.example.com')
print(response.text)

  That demonstrates the issue i.e a small piece of code which reproduces the issue and can be run
  with out any other (or as few as possible) external dependencies. Issues without this may will
  likely be rejected without much consideration. Often this can be as simple as importing a package,
  if this is a packaging issue, try that first.

  Pointers to repositories with usage of `pipenv` are very welcome, archives with examples are too,
  e.g. if a certain package structure is needed. This must be only source code, binaries are not
  used ever by us. But beware, that e.g. we cannot click around and stuff. Still do an effort to
  make the usage obvious. Having a compile script in the repo is perfect.

  But this cannot be much code for non-commercial users, since that causes too much effort. We
  cannot just compile your code, run it and have it download a control and command center.

- Provide in your issue the Nuitka options used
  
  python -m nuitka --standalone --lto=yes --include-plugin-directory=C:/Users/Administrator/Desktop/proje/venv/Lib/site-packages/requests --output-dir=output_dir test.py

  Ideally use the `# nuitka-project:` options feature in the code, so options and example code go
  along. Alternatively state the command line.

  [Nuitka Options in the code](https://nuitka.net/doc/user-manual.html#nuitka-options-in-the-code)

- Avoid unnecessary options

  Do not use `--deployment` as it disables all potential bug catching, we are not here to replace
  what it does, remove that option the moment you face an issue and report only then.

  Do not use `--onefile` if the issue also happens with `--standalone`. Minimize the use of options
  as far as you can, please. Do not disable outputs with `--quiet` and do not disable warnings.

- Note if this is a regression

  If it used to work in an earlier version of Nuitka, please note what you know about that. Since
  git bisect is a thing for which we do a lot of error to make it usable, this will help
  dramatically to isolate the issue.

- Consider getting commercial support

  [Nuitka commercial](https://nuitka.net/doc/commercial.html) offers subscriptions and priority
  support. This will accelerate your problem solution and helps to sustain Nuitka development. Also
  you then have a chance to provide source code of your project to us, which might simplify things,
  or pay for time to solve your issues in your environment.

Some things are not welcome, please consider it.

- Do *not* post screenshots

  These are not welcome unless absolutely necessary, e.g. because of Qt display problem, instead
  capture the output of programs, so things are searchable and copy&paste will work. I just plainly
  don't want to manually copy strings and hope they match.

- Do *not* close the issue yourself, we will close things on stable releases

  We close issues only when they are released as a stable version, e.g. in a hotfix or a new
  release, before that it will be "Done" in planning and go through `factory` and `develop` tags to
  indicate they are solved there.

  Of course, if you find out your issue is invalid, please do close it, and we then attach the
  `invalid` tag.

- Do *not* report against factory version

  Unless you were asked to test it there, it is frequently very broken, and there is only noise to
  be had. Telling me about it on Discord would be a better idea.

- Do *not* let this template remain part of the issue, it's noise.

The error output I received:

Nuitka-Options: Used command line options: --standalone --lto=yes
Nuitka-Options: --include-plugin-directory=C:/Users/Administrator/Desktop/proje/venv/Lib/site-packages/requests
Nuitka-Options: --output-dir=output_dir test.py
Nuitka: Starting Python compilation with Nuitka '2.4.8' on Python '3.9' commercial grade 'not installed'.
PASS 1: 0.0%|                         | 0/19, requests.utilsTraceback (most recent call last):
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\__main__.py", line 206, in <module>
    main()
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\__main__.py", line 188, in main
    MainControl.main()
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\MainControl.py", line 1155, in main
    _main()
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\MainControl.py", line 995, in _main
    main_module = _createMainModule()
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\MainControl.py", line 256, in _createMainModule
    optimizeModules(main_module.getOutputFilename())
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 332, in optimizeModules
    finished = makeOptimizationPass()
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 288, in makeOptimizationPass
    changed, micro_passes = optimizeModule(current_module)
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 166, in optimizeModule
    return optimizeCompiledPythonModule(module)
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 126, in optimizeCompiledPythonModule
    considerUsedModules(module=module, pass_count=pass_count)
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\importing\Recursion.py", line 521, in considerUsedModules
    new_module = recurseTo(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\importing\Recursion.py", line 80, in recurseTo
    module = _recurseTo(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\importing\Recursion.py", line 38, in _recurseTo
    module = Building.buildModule(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\tree\Building.py", line 1434, in buildModule
    module = _createModule(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\tree\Building.py", line 1134, in _createModule
    result = CompiledPythonPackage(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\nodes\ModuleNodes.py", line 712, in __init__
    CompiledPythonModule.__init__(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\nodes\ModuleNodes.py", line 294, in __init__
    self.locals_scope = getLocalsDictHandle(
  File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\nodes\LocalsScopes.py", line 41, in getLocalsDictHandle
    raise NuitkaOptimizationError(
nuitka.Errors.NuitkaOptimizationError:
duplicate locals name
globals_requests
module_dict
requests
C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\requests\__init__.py
requests
C:\Users\Administrator\Desktop\proje\venv\Lib\site-packages\requests\__init__.py

The above information should be included in a bug report.
Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'. Please include it in your bug
Nuitka-Reports: report.
PASS 1: 0.0%|                         | 0/19, requests.utils

nuitka-crash-report.xml output:

<nuitka-compilation-report nuitka_version="2.4.8" nuitka_commercial_version="not installed" completion="exception">
<exception exception_type="NuitkaOptimizationError" exception_value=" duplicate locals name globals_requests module_dict requests C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\requests\__init__.py requests C:\Users\Administrator\Desktop\proje\venv\Lib\site-packages\requests\__init__.py The above information should be included in a bug report."> Traceback (most recent call last): File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\MainControl.py", line 1155, in main _main() File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\MainControl.py", line 995, in _main main_module = _createMainModule() File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\MainControl.py", line 256, in _createMainModule optimizeModules(main_module.getOutputFilename()) File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 332, in optimizeModules finished = makeOptimizationPass() File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 288, in makeOptimizationPass changed, micro_passes = optimizeModule(current_module) File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 166, in optimizeModule return optimizeCompiledPythonModule(module) File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\optimizations\Optimization.py", line 126, in optimizeCompiledPythonModule considerUsedModules(module=module, pass_count=pass_count) File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\importing\Recursion.py", line 521, in considerUsedModules new_module = recurseTo( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\importing\Recursion.py", line 80, in recurseTo module = _recurseTo( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\importing\Recursion.py", line 38, in _recurseTo module = Building.buildModule( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\tree\Building.py", line 1434, in buildModule module = _createModule( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\tree\Building.py", line 1134, in _createModule result = CompiledPythonPackage( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\nodes\ModuleNodes.py", line 712, in __init__ CompiledPythonModule.__init__( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\nodes\ModuleNodes.py", line 294, in __init__ self.locals_scope = getLocalsDictHandle( File "C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\nuitka\nodes\LocalsScopes.py", line 41, in getLocalsDictHandle raise NuitkaOptimizationError( nuitka.Errors.NuitkaOptimizationError: duplicate locals name globals_requests module_dict requests C:\Users\Administrator\Desktop\proje\venv\lib\site-packages\requests\__init__.py requests C:\Users\Administrator\Desktop\proje\venv\Lib\site-packages\requests\__init__.py The above information should be included in a bug report. </exception>
<module name="requests.utils" kind="CompiledPythonModule" usage="root_module" reason="Root module" source_path="${sys.prefix}\Lib\site-packages\requests\utils.py" distribution="requests">
<module_usages>
<module_usage name="codecs" finding="absolute" line="9"/>
<module_usage name="contextlib" finding="absolute" line="10"/>
<module_usage name="io" finding="absolute" line="11"/>
<module_usage name="os" finding="absolute" line="12"/>
<module_usage name="re" finding="absolute" line="13"/>
<module_usage name="socket" finding="absolute" line="14"/>
<module_usage name="struct" finding="absolute" line="15"/>
<module_usage name="sys" finding="absolute" line="16"/>
<module_usage name="tempfile" finding="absolute" line="17"/>
<module_usage name="warnings" finding="absolute" line="18"/>
<module_usage name="zipfile" finding="absolute" line="19"/>
<module_usage name="collections" finding="absolute" line="20"/>
<module_usage name="collections.OrderedDict" finding="not-found" line="20"/>
<module_usage name="urllib3.util" finding="absolute" line="22"/>
<module_usage name="urllib3.util.make_headers" finding="not-found" line="22"/>
<module_usage name="urllib3.util.parse_url" finding="not-found" line="22"/>
<module_usage name="requests" finding="absolute" line="24"/>
<module_usage name="requests.certs" finding="relative" line="24"/>
<module_usage name="requests.__version__" finding="relative" line="25"/>
<module_usage name="requests._internal_utils" finding="relative" line="28"/>
<module_usage name="requests.compat" finding="relative" line="34"/>
<module_usage name="requests.compat" finding="relative" line="42"/>
<module_usage name="requests.compat" finding="relative" line="43"/>
<module_usage name="requests.cookies" finding="relative" line="52"/>
<module_usage name="requests.exceptions" finding="relative" line="53"/>
<module_usage name="requests.structures" finding="relative" line="59"/>
<module_usage name="winreg" finding="absolute" line="78"/>
<module_usage name="netrc" finding="absolute" line="214"/>
<module_usage name="ntpath" finding="absolute" line="220"/>
<module_usage name="ntpath" finding="absolute" line="279"/>
<module_usage name="ntpath" finding="absolute" line="281"/>
<module_usage name="ntpath" finding="absolute" line="297"/>
</module_usages>
</module>
<performance>
<memory_usage name="after_launch" value="24129536"/>
</performance>
<data_composer blob_size="None"/>
<command_line>
<option value="--standalone"/>
<option value="--lto=yes"/>
<option value="--include-plugin-directory=C:/Users/Administrator/Desktop/proje/venv/Lib/site-packages/requests"/>
<option value="--output-dir=output_dir"/>
<option value="test.py"/>
</command_line>
<plugins>
<plugin name="anti-bloat" user_enabled="no"/>
<plugin name="data-files" user_enabled="no"/>
<plugin name="delvewheel" user_enabled="no"/>
<plugin name="dll-files" user_enabled="no"/>
<plugin name="eventlet" user_enabled="no"/>
<plugin name="gevent" user_enabled="no"/>
<plugin name="gi" user_enabled="no"/>
<plugin name="glfw" user_enabled="no"/>
<plugin name="implicit-imports" user_enabled="no"/>
<plugin name="kivy" user_enabled="no"/>
<plugin name="matplotlib" user_enabled="no"/>
<plugin name="multiprocessing" user_enabled="no"/>
<plugin name="options-nanny" user_enabled="no"/>
<plugin name="pbr-compat" user_enabled="no"/>
<plugin name="pkg-resources" user_enabled="no"/>
<plugin name="pywebview" user_enabled="no"/>
<plugin name="spacy" user_enabled="no"/>
<plugin name="transformers" user_enabled="no"/>
</plugins>
<distributions>
<distribution name="requests" version="2.32.3" installer="pip"/>
</distributions>
<python python_exe="${sys.prefix}\Scripts\python.exe" python_flavor="CPython Official" python_version="3.9.0" os_name="Windows" os_release="10" arch_name="x86" filesystem_encoding="utf-8">
<search_path>
<path value="${cwd}"/>
<path value="${sys.real_prefix}\DLLs"/>
<path value="${sys.real_prefix}\lib"/>
<path value="${sys.real_prefix}"/>
<path value="${sys.prefix}"/>
<path value="${sys.prefix}\lib\site-packages"/>
</search_path>
</python>
<output run_filename="${cwd}\output_dir\test.dist\test.exe"/>
</nuitka-compilation-report>

**Comment by enescantas** [comment]
@kayhayen Did I provide incomplete information?

**Comment by KRRT7** [comment]
https://github.com/KRRT7/Nuitka-Issues/actions/runs/10675356737/job/29587222195
it builds correctly for me on all python versions [jobs > build with nuitka > look at the output there]

**Comment by KRRT7** [comment]
` --include-plugin-directory=C:/Users/Administrator/Desktop/proje/venv/Lib/site-packages/requests`
looks very wrong, remove it and try again.

**Comment by kayhayen** [comment]
User error clearly
